addon/components/infinity-loader.js

```diff
@@ -111,44 +63,38 @@ const InfinityLoaderComponent = Ember.Component.extend({
     ) {
       return false;
     }
-    return this._bottomOfScrollableOffset() > this._triggerOffset();
+
+    this._debounceScrolledToBottom();
   },
 
-  _loadMoreIfNeeded() {
-    const infinityModel = this.get('infinityModelContent');
-    if (this._shouldLoadMore() && infinityModel) {
-      this.sendAction('loadMoreAction', infinityModel);
-    }
+  didExitViewport() {
+    this._cancelTimers();
   },
 
-  _setupScrollable() {
-    var scrollable = this.get('scrollable');
-    if (Ember.typeOf(scrollable) === 'string') {
-      var items = Ember.$(scrollable);
-      if (items.length === 1) {
-        this.set('_scrollable', items.eq(0));
-      } else if (items.length > 1) {
-        throw new Ember.Error("Ember Infinity: Multiple scrollable elements found for: " + scrollable);
-      } else {
-        throw new Ember.Error("Ember Infinity: No scrollable element found for: " + scrollable);
-      }
-      this.set('_customScrollableIsDefined', true);
-    } else if (scrollable === undefined || scrollable === null) {
-      this.set('_scrollable', Ember.$(window));
-      this.set('_customScrollableIsDefined', false);
-    } else {
-      throw new Ember.Error("Ember Infinity: Scrollable must either be a css selector string or left empty to default to window");
-    }
+  _scheduleScrolledToBottom() {
+    this._schedulerTimer = run.scheduleOnce('afterRender', this, this._debounceScrolledToBottom);
   },
 
-  loadedStatusDidChange: Ember.observer('infinityModelContent.reachedInfinity', 'destroyOnInfinity', function () {
-    if (this.get('infinityModelContent.reachedInfinity') && this.get('destroyOnInfinity')) {
-      this.destroy();
+  _debounceScrolledToBottom() {
+    /*
+     This debounce is needed when there is not enough delay between onScrolledToBottom calls.
+     Without this debounce, all rows will be rendered causing immense performance problems
+     */
+    function loadMore() {
+      this.sendAction('loadMoreAction', this.get('infinityModelContent'));
     }
-  }),
+    this._debounceTimer = run.debounce(this, loadMore, this.get('eventDebounce'));
+  },
+
+  _cancelTimers() {
+    run.cancel(this._schedulerTimer);
+    run.cancel(this._debounceTimer);
+  },
 
-  infinityModelPushed: Ember.observer('infinityModelContent.length', function() {
-    Ember.run.scheduleOnce('afterRender', this, this._loadMoreIfNeeded);
+  infinityModelPushed: observer('infinityModelContent.length', 'viewportEntered', function() {
```

Comment:
I'm thinking method this is doubling down on calling the loadMoreAction.  If we did enter the viewport, the `didEnterViewport` method will be called and will fetch more.

I'm thinking this should be removed.

Reply:
Actually I think I am wrong.  If rows are removed, then we want to check again if the `infinity-loader` is now in viewport.